src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -1264,8 +1264,17 @@ internal new NamedTypeSymbol GetSpecialType(SpecialType specialType)
                 throw new ArgumentOutOfRangeException(nameof(specialType), $"Unexpected SpecialType: '{(int)specialType}'.");
             }
 
-            var result = Assembly.GetSpecialType(specialType);
-            Debug.Assert(result.SpecialType == specialType);
+            NamedTypeSymbol result;
+            if (IsTypeMissing((WellKnownType)specialType))
```

Comment:
>(WellKnownType)specialType [](start = 30, length = 26)

Rather than casting between unrelated enum types, we should add overloads for IsTypeMissing/MakeTypeMissing that take SpecialType